backend/models/group.js

```diff
@@ -255,56 +242,51 @@ function getObjectsArrayAsIfcGuids(data) {
 	});
 }
 
-async function ifcGuidsToUUIDs(account, model, branch, revId, ifcGuids) {
-	if (!ifcGuids || ifcGuids.length === 0) {
-		return Promise.resolve([]);
-	}
-
-	const query = {"metadata.IFC GUID": { $in: ifcGuids } };
-	const project = { parents: 1, _id: 0 };
-
-	const sceneCollName = model + ".scene";
-	const results = await db.find(account, sceneCollName, query, project);
-
-	if (results.length === 0) {
-		return [];
-	}
-
-	const history = await History.getHistory({ account, model }, branch, revId);
-	const parents = results.map(x => x = x.parents).reduce((acc, val) => acc.concat(val), []);
-
-	const meshQuery = { _id: { $in: history.current }, shared_id: { $in: parents }, type: "mesh" };
-	const meshProject = { shared_id: 1, _id: 0 };
-
-	return db.find(account, sceneCollName, meshQuery, meshProject);
-}
-
 const Group = {};
 
 Group.create = async function (account, model, branch = "master", rid = null, sessionId, creator = "", data) {
 	const newGroup = Object.assign({}, data);
 
 	const convertedObjects = await getObjectsArrayAsIfcGuids(data, false);
-	const allowedFields = ["description", "name", "objects","rules","color","transformation","issue_id","risk_id","view_id"];
 
 	let typeCorrect = (!data.objects !== !data.rules);
 
-	allowedFields.forEach((key) => {
-		if (fieldTypes[key] && utils.hasField(data, key)) {
+	Object.keys(data).forEach((key) => {
+		if (fieldTypes[key]) {
 			if (utils.typeMatch(data[key], fieldTypes[key])) {
-				if (key === "objects" && data.objects) {
-					newGroup.objects = cleanEmbeddedObject(key, convertedObjects);
-				} else if (key === "color") {
-					newGroup[key] = data[key].map((c) => parseInt(c, 10));
-				} else if (key === "transformation") {
-					newGroup[key] = data[key];
-				} else {
-					if (key === "rules"
-						&& data.rules
-						&& !checkRulesValidity(data.rules)) {
-						typeCorrect = false;
-					}
-					newGroup[key] = cleanEmbeddedObject(key, data[key]);
+				switch (key) {
+					case "description":
+						newGroup[key] = data[key];
+						break;
+					case "name":
+						newGroup[key] = data[key];
+						break;
+					case "objects":
+						if (data.objects) {
+							newGroup.objects = cleanEmbeddedObject(key, convertedObjects);
+						}
+						break;
+					case "rules":
+						if (data.rules && !checkRulesValidity(data.rules)) {
+							typeCorrect = false;
+						}
+						newGroup[key] = cleanEmbeddedObject(key, data[key]);
+						break;
+					case "color":
+						newGroup[key] = data[key].map((c) => parseInt(c, 10));
+						break;
+					case "transformation":
```

Comment:
Should these just be `default`?

Persumably we already prune out any values that are not recognised at L255